shell/browser/api/electron_api_session.h

```diff
@@ -55,6 +56,29 @@ class ElectronBrowserContext;
 
 namespace api {
 
+class WebContents;
+
+#if BUILDFLAG(ENABLE_ELECTRON_EXTENSIONS)
+class ExtensionTabDetails {
+ public:
+  ExtensionTabDetails();
+  ExtensionTabDetails(const ExtensionTabDetails& other);
+  ~ExtensionTabDetails();
```

Comment:
Since this class uses default constructor and copy constructor, there is no need defining them, you can just do:

```c++
struct ExtensionTabDetails {
  int window_id = 0;
  ...
  std::string muted_extension_id;
};
```

Reply:
@zcbenz Then I get
``` 
../../electron\shell/browser/api/electron_api_session.h(62,1): error: [chromium-style] Complex class/struct needs an explicit out-of-line constructor.
```

Reply:
@zcbenz Then I get
```
../../electron\shell/browser/api/electron_api_session.h(62,1): error: [chromium-style] Complex class/struct needs an explicit out-of-line constructor.
struct ExtensionTabDetails {
^
```

Reply:
I forgot Chromium's restriction, then it is fine as it is.